one_fm/operations/doctype/shift_permission/shift_permission.py

```diff
@@ -13,12 +13,14 @@ def validate(self):
 		self.check_permission_type()
 		self.check_shift_details_value()
 		self.validate_date()
+
+	def on_update(self):
```

Comment:
You can directly use the `validate` controller method instead of validating inside the `on_update` and carry out any validation inside that method.